.github/workflows/build.yml

```diff
@@ -19,32 +19,60 @@ jobs:
       - uses: actions/checkout@v4
       - uses: gradle/wrapper-validation-action@v2.1.1
 
-  gradle:
+  build:
     runs-on: ubuntu-latest
     steps:
 
       - name: Checkout
         uses: actions/checkout@v4
 
-      - name: Gradle build, test and check
+      - name: Gradle build without tests
         uses: ./.github/workflows/gradle-goal
         with:
-          command: "./gradlew check"
+          # We also compile the test-classes, even though we are skipping the tests
+          command: "./gradlew build testClasses -x test"
 
-      - name: Warmup gradle wrapper
-        uses: ./.github/workflows/gradle-goal
-        with:
-          command: "./gradlew check"
-      - name: Store test results
-        if: success() || failure()
+      # We first tar the working directory and then upload it
+      # We do this because uploading directories with many files directly as artifacts is very slow
+      - name: Create working dir archive
+        run: "mkdir ./wdarch && tar --exclude='./wdarch' -czf ./wdarch/working-dir-build-cache.tar.gz ."
+      - name: Cache working directory with build results
```

Comment:
Maybe `v4` can be faster? See https://github.com/actions/upload-artifact?tab=readme-ov-file#usage

In this case you can case use v4 since it's not a matrix job uploading artifacts

Reply:
Just did a test, v4 seems to be significantly faster. It is even faster than my tar solution:
![image](https://github.com/elastic/elastic-otel-java/assets/6066661/266a5df3-79fa-450a-a751-87a3fbb5da23)

I'll update the PR ans switch to the `@v4` variants

Reply:
Unfortunatly the `upload-artifact` action doesn't preserve permissions:
```
/home/runner/work/_temp/951c9356-ece7-4462-9b6c-2985a0d9fb9b.sh: line 1: ./gradlew: Permission denied
```

As suggested by the [readme](https://github.com/actions/upload-artifact?tab=readme-ov-file#permission-loss), I'll stay with taring but change the comment explaining the reasoning

Reply:
Oh dear! I missed that. Tath's a pain in the heck

Reply:
I wonder if you only need to store the `build` folder? If so, will the permissions won't be a problem?

Reply:
You would need to store the outputs of every subproject which usually should go into the build folder. In addition I think gradle stores the hashes of the build contents somewhere, but I'm not sure about where.

Also, you'd need to checkout again and merge things back. I think just uploading the entire working dir and downloading it again is a lot simpler and less error-prone than checking out the repo again and trying to merge things back together.